lib/api/objectPut.js

```diff
@@ -190,95 +179,71 @@ function objectPut(authInfo, request, streamingV4Params, log, callback) {
-                                        streamingV4Params, log,
-                                            (err, contentMD5) =>
-                                                callback(err, contentMD5,
-                                                    corsHeaders));
-                                });
-                        }
-                        return _storeIt(bucket, objectKey, objMD,
-                                        authInfo, canonicalID, null, request,
-                                        streamingV4Params, log,
-                                (err, contentMD5) =>
-                                    callback(err, contentMD5, corsHeaders));
-                    });
-        }
-        if (serverSideEncryption) {
-            return kms.createCipherBundle(
-                serverSideEncryption,
-                log, (err, cipherBundle) => {
-                    if (err) {
-                        return callback(errors.InternalError, null,
-                            corsHeaders);
-                    }
-                    return _storeIt(bucket, objectKey, objMD,
-                                    authInfo, canonicalID, cipherBundle,
-                                    request, streamingV4Params, log,
-                            (err, contentMD5) =>
-                                callback(err, contentMD5, corsHeaders));
-                });
-        }
-        return _storeIt(bucket, objectKey, objMD, authInfo, canonicalID,
-                        null, request, streamingV4Params, log,
-                        (err, contentMD5) =>
-                            callback(err, contentMD5, corsHeaders));
+        return async.waterfall([
+            function handleTransientOrDeleteBuckets(next) {
+                if (bucket.hasTransientFlag() || bucket.hasDeletedFlag()) {
+                    return cleanUpBucket(bucket, canonicalID, log, next);
+                }
+                return next();
+            },
+            function createCipherBundle(next) {
+                const serverSideEncryption = bucket.getServerSideEncryption();
+                if (serverSideEncryption) {
+                    return kms.createCipherBundle(
+                            serverSideEncryption, log, next);
+                }
+                return next(null, null);
+            },
+            function objectCreateAndStore(cipherBundle, next) {
+                return createAndStoreObject(bucketName,
+                bucket, objectKey, objMD, authInfo, canonicalID, cipherBundle,
+                request, streamingV4Params, log, next);
+            },
+        ], (err, res) => {
+            if (err) {
+                return callback(err, responseHeaders);
+            }
+            const newByteLength = request.parsedContentLength;
+            // Utapi expects null or a number for oldByteLength
+            const oldByteLength = objMD && objMD['content-length']
```

Comment:
as discussed, since the object or version we would have overwritten (if we overwrite any) is the master version, this is the correct objMD to use to calculate the old byte length here.  I added a comment to help